internal/pkg/levee/levee.go

```diff
@@ -78,6 +104,76 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	return nil, nil
 }
 
+// Use the EAR pointer analysis as the propagation engine
+func runEAR(pass *analysis.Pass, state *earpointer.Partitions) (interface{}, error) {
+	conf, err := config.ReadConfig()
+	if err != nil {
+		return nil, err
+	}
+	funcSources := pass.ResultOf[source.Analyzer].(source.ResultType)
+	taggedFields := pass.ResultOf[fieldtags.Analyzer].(fieldtags.ResultType)
+	suppressedNodes := pass.ResultOf[suppression.Analyzer].(suppression.ResultType)
+	// Return whether a field is tainted.
+	isTaintField := func(named *types.Named, index int) bool {
+		if tt, ok := named.Underlying().(*types.Struct); ok {
+			return conf.IsSourceField(utils.DecomposeField(named, index)) ||
+				taggedFields.IsSourceField(tt, index)
+		}
+		return false
+	}
+	// Collect all the taint sources
+	for fn, sources := range funcSources {
+		// Transitively get the set of functions called within "fn".
+		// This set is used to narrow down the set of references needed to be
+		// considered during EAR heap traversal. It can also help reducing the
+		// false positives.
+		callees := make(map[*ssa.Function]bool)
+		callees[fn] = true
+		earpointer.GetCalleeFunctions(fn, callees, 7)
```

Comment:
Now it is a configurable number. It will be used in a broader context in the future PRs. Especially, currently we limit the search scope to the callees within the function which defines the taint source, e.g. 

```
func f1(a *T) {
   taintf(a)  // taint sink
}

func g1(a *T) {
   f1(a)
}

func h1(src *T) {   // taint source 
   g1(src)
}
```

This constraint limits the scope to h1(), g1(), and f1() only. However, another important case is: 
```
func f2() string {
   a := new T{} //  taint source
   return fmt.Printf(a)
}

func g2() {
   return f2(a)
}

func h2() {   // taint source 
   x := g2()
  sinkf(x) // taint sink
}
```
That is, we need to go from f2() to its callers g2() and h2(). We may need to limit how much further we shall go, and this "depth" can be used to control that. In future PRs I may rename it to "CallChainSpan". :)